### ISSUE TITLE
Why is there no overlap between the test set and the validation set? Isn't the test set just 1000 records extracted from the validation set?

### ISSUE THREAD

**Issue by liumc14** [meaningful96/CoopRAG#4]
> No. The “1,000 samples” mentioned in the paper are not the training set; they are the test (evaluation) set described in Section 4.1. In particular, the paper evaluates on 1,000 test questions from each multi-hop dataset (HotpotQA, MuSiQue, and 2Wiki), and it reuses the same test set used by HippoRAG to ensure a fair comparison against existing baselines. 
> 
> The retriever training data is constructed separately and is not formed by “starting from 1,000 samples and expanding them back into 1,000 training samples.” As described in Section 3.6, for retriever fine-tuning the authors first apply question unrolling to questions in the training split to obtain unrolled questions, then pair each unrolled question with one positive document and one hard negative document (following the ColBERT setup) and train the retriever using an InfoNCE-style contrastive objective. 
> 
> We also applied the same question-unrolling preprocessing to the training data, and we have already updated the processed version on [HuggingFace: meaningful96/CoopRAG](meaningful96/CoopRAG). 

 _Originally posted by @meaningful96 in [#3](https://github.com/meaningful96/CoopRAG/issues/3#issuecomment-3691934188)_

I noticed in your paper that the test set follows the method used in Hipporaprag. I also checked the Hipporaprag paper, and it also follows the method used in Ircot to extract the test set, i.e., randomly sampling from the validation set. However, I found that the test set and validation set in your published dataset do not overlap. Why is that? I look forward to your reply. Thank you.
<img width="724" height="256" alt="Image" src="https://github.com/user-attachments/assets/d2e51729-242c-471e-892e-eaa9c53b3288" />

<img width="1104" height="252" alt="Image" src="https://github.com/user-attachments/assets/eedb06c5-2a8c-4761-ab47-8c4fe98c5e85" />

<img width="1078" height="181" alt="Image" src="https://github.com/user-attachments/assets/6e190470-af73-4009-a787-67548c1f6174" />

**Comment by meaningful96** [comment]
Thank you for your question. I may have misunderstood the exact point of your concern, so let me clarify both possible interpretations.

If your question refers to the dataset I released on Hugging Face—specifically, that the validation and test sets do not overlap—this is intentional. The validation and test sets must be disjoint to avoid data leakage. Allowing overlap would compromise the reliability of the evaluation.

On the other hand, if your question is about why our test set does not exactly match the one provided by HippoRAG, our dataset was based on the one provided by the GFM-RAG repository. As mentioned in their paper, they used 1,000 samples for fair comparison with HippoRAG. Our experiments followed the same setup: we reproduced all baselines using those same 1,000 queries to ensure fairness. Therefore, although we downloaded the data from GFM-RAG, its origin traces back to the HippoRAG dataset, which is why we cited it accordingly.

**Comment by liumc14** [comment]
> Thank you for your question. I may have misunderstood the exact point of your concern, so let me clarify both possible interpretations.
> 
> If your question refers to the dataset I released on Hugging Face—specifically, that the validation and test sets do not overlap—this is intentional. The validation and test sets must be disjoint to avoid data leakage. Allowing overlap would compromise the reliability of the evaluation.
> 
> On the other hand, if your question is about why our test set does not exactly match the one provided by HippoRAG, our dataset was based on the one provided by the GFM-RAG repository. As mentioned in their paper, they used 1,000 samples for fair comparison with HippoRAG. Our experiments followed the same setup: we reproduced all baselines using those same 1,000 queries to ensure fairness. Therefore, although we downloaded the data from GFM-RAG, its origin traces back to the HippoRAG dataset, which is why we cited it accordingly.

Is my understanding correct: In your publicly available dataset, both the training and validation sets are derived from the training set split of the original multi-hop question answering dataset. The test set, like in Hipporag, is sampled from the validation set of the original multi-hop question answering dataset.

**Comment by meaningful96** [comment]
Yes, your understanding is correct.

In our publicly released dataset, the training set and validation set are both constructed from the training split of the original multi-hop QA dataset. This design choice allows us to tune models without ever touching the official validation data of the original benchmark.

Meanwhile, the test set is created following the same protocol as HippoRAG and GFM-RAG: it is sampled from the original validation split of the multi-hop QA dataset. This ensures that our evaluation is directly comparable with prior work, while maintaining a clean separation between development data and evaluation data within our released dataset.

So to summarize:
- Original training split → our train / our validation
- Original validation split → our test

This guarantees both fair comparison with previous methods and strict separation between validation and test data in our benchmark.
